std/mmfile.d

```diff
@@ -134,19 +134,25 @@ class MmFile
             char c = 0;
             core.sys.posix.unistd.write(fd, &c, 1);
         }
-        else if (prot & PROT_READ && size == 0)
-            size = statbuf.st_size;
+        else if (prot & PROT_READ && size == 0) {
+            if(statbuf.st_size > 0) size = statbuf.st_size;
+            else return;
+        }
         this.size = size;
 
         // Map the file into memory!
         size_t initial_map = (window && 2*window<size)
             ? 2*window : cast(size_t) size;
-        auto p = mmap(address, initial_map, prot, flags, fd, 0);
-        if (p == MAP_FAILED)
-        {
-            errnoEnforce(false, "Could not map file into memory");
+        if(initial_map > 0) {
```

Comment:
Is that necessary now that there is an early return?

Reply:
Not in the readonly case. I'll revert this so that we can focus on read mode